### PR TITLE
RCAL-246: Add assign_wcs step

### LINKED ISSUE #353: Implement a basic wcs object in L2 products
_Issue [RCAL-246](https://jira.stsci.edu/browse/RCAL-246) was created on JIRA by [Nadia Dencheva](https://jira.stsci.edu/secure/ViewProfile.jspa?name=dencheva):_

### PR DESCRIPTION
**Pull request by nden**
Closes #353
Resolves RCAL-246

**Description**

This PR adds an `assign_wcs` step to romancal.

Fixed formatting of `Changes.rst`.

Checklist
- [x] Tests

- [x] Documentation

- [x] Change log

- [x] Milestone

- [x] Label(s)

### CODE REVIEW

**Review by ddavis-stsci — APPROVED** [review]
I don't see any issues.

File: romancal/assign_wcs/assign_wcs_step.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    """ Create a gWCS object and store it in ``Model.meta``.
+
+    Parameters
+    ----------
+    dmodel : `~roman_datamodels.datamodels.WfiImage`
+        The exposure.
+    reference_files : dict
+        A dict {reftype: reference_file_name} containing all
+        reference files that apply to this exposure.
+    """
```
Comment (dmggh) [inline romancal/assign_wcs/assign_wcs_step.py:55]: Could include `return` section in epydoc for updated dmodel.

File: romancal/assign_wcs/pointing.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    at the center of the aperture, ROLL_REF in deg.
+
+    Parameters
+    ----------
+    input_model : `roman_datamodels.WfiImage`
+        Roman imaging exposure data model.
+    wrap_v2_at : float
+        At what angle to wrap V2. [deg]
+    wrap_lon_at : float
+        At what angle to wrap logitude. [deg]
```
Comment (dmggh) [inline romancal/assign_wcs/pointing.py:21]: Could include `return' section in epydoc for m.

File: romancal/assign_wcs/assign_wcs_step.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+class AssignWcsStep(RomanStep):
+    """ Assign a gWCS object to a science image.
+    """
+
+    reference_file_types = []
+
+    def process(self, input_model):
+        reference_file_names = {}
+        with rdm.open(input_model) as input_model:
```
Comment (PaulHuwe) [inline romancal/assign_wcs/assign_wcs_step.py:34]: Style question: do we want to open the model name as the same name?

**Review by dmggh — APPROVED** [review]
Minor docs comments. Looks good to me.

File: romancal/assign_wcs/assign_wcs_step.py [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+            for reftype in self.reference_file_types:
+                log.info(f'reftype, {reftype}')
+                reffile = self.get_reference_file(input_model, reftype)
+                reference_file_names[reftype] = reffile if reffile else ""
+            log.debug(f'reference files used in assign_wcs: {reference_file_names}')
+            input_model = self.open_model(input_model)
```
Comment (PaulHuwe) [inline romancal/assign_wcs/assign_wcs_step.py:40]: Opening the same model a second time?

File: romancal/assign_wcs/pointing.py [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+from astropy.modeling.models import RotationSequence3D, Scale
+from gwcs.geometry import SphericalToCartesian, CartesianToSpherical
+
+
+def v23tosky(input_model, wrap_v2_at=180, wrap_lon_at=360):
+    """ Create the transform from telescope to sky.
+
+    The transform is defined with a reference point in a Frame
+    associated tih the telescope (V2, V3) in arcsec, the corresponding
+    reference poiont on sky (RA_REF, DEC_REF) i deg, and the position angle
```
Comment (PaulHuwe) [inline romancal/assign_wcs/pointing.py:11]: "in deg"

**Review by PaulHuwe — CHANGES_REQUESTED** [review]

File: romancal/assign_wcs/assign_wcs_step.py (hunk 1 above)
Comment (nden) [reply romancal/assign_wcs/assign_wcs_step.py:60]: done

File: romancal/assign_wcs/pointing.py (hunk 2 above)
Comment (nden) [reply romancal/assign_wcs/pointing.py:21]: done

File: romancal/assign_wcs/assign_wcs_step.py (hunk 4 above)
Comment (nden) [reply romancal/assign_wcs/assign_wcs_step.py]: Thanks for catching this!

File: romancal/assign_wcs/assign_wcs_step.py (hunk 3 above)
Comment (nden) [reply romancal/assign_wcs/assign_wcs_step.py]: there's no rule against this

**Comment by nden** [comment]
Comments were addressed, feel free to merge when tests pass

File: romancal/assign_wcs/assign_wcs_step.py (hunk 3 above)
Comment (PaulHuwe) [reply romancal/assign_wcs/assign_wcs_step.py]: Hai, is a question of style. Resolving this for merge.

**Review by PaulHuwe — APPROVED** [review]

**Comment by PaulHuwe** [comment]
Tests passed - merging.

**Pull request merged**